chunky/src/java/se/llbit/chunky/ui/ChunkMap.java

```diff
@@ -337,12 +340,18 @@ public void renderView(File targetFile, ProgressTracker progress) {
   }
 
   @Override public void viewUpdated(ChunkView view) {
-    synchronized (this) {
-      this.view = view;
-      mapBuffer.updateView(view);
+    if (!viewUpdateScheduled) {
+      viewUpdateScheduled = true;
+      executor.schedule(() -> {
+        lastUpdate = System.nanoTime();
+        viewUpdateScheduled = false;
+
+        this.view = view;
+        mapBuffer.updateView(view);
+        mapBuffer.redrawView(mapLoader, chunkSelection);
+        repaintDeferred();
+      }, (1000000000/VIEW_MAX_UPDATES) - (System.nanoTime() - lastUpdate), TimeUnit.NANOSECONDS);
```

Comment:
```suggestion
      }, (1_000_000_000 / VIEW_MAX_UPDATES) - (System.nanoTime() - lastUpdate), TimeUnit.NANOSECONDS);
```

(after counting the zeros twice before realizing that this is 10^9)